### PR TITLE
Auto assign hotcue color

### PR DESCRIPTION
**Pull request by Holzhaus**
This adds the ability to assign hotcue colors automatically (based on the index), similar to what Serato does.

Currently missing:
- [x] ~~Translations~~

### CODE REVIEW

**Comment by Holzhaus** [comment]
By the way, I'm unsure what the preferred naming scheme for control objects is. Is this documented somewhere? I see both `PascalCase` and `lowercase_with_underscores` being used.

**Comment by uklotzde** [comment]
As far as I know we must not update the translation files locally. Those generated files are fetched from Transifex.

**Comment by Holzhaus** [comment]
> As far as I know we must not update the translation files locally. Those generated files are fetched from Transifex.

I followed step 1-4 of the [procedure described in wiki](https://www.mixxx.org/wiki/doku.php/internationalization#updating_translation_templates) to update the translation templates for Transifex.

**Comment by daschuer** [comment]
Yes, @uklotzde is right. We updated the translation templates directly from the master branch. 
Doing it in feature branches may introduce merge conflicts. 

Can you please remove the last commit and force push? 

git reset --hard HEAD^1
git push -f

**Comment by Holzhaus** [comment]
@daschuer Ok, done.

File: src/preferences/dialog/dlgprefdeckdlg.ui [hunk 1]
```diff
@@ -33,6 +33,16 @@
         </property>
        </widget>
       </item>
+      <item row="4" column="0">
+       <widget class="QLabel" name="labelAutoHotcueColors">
+        <property name="text">
+          <string>Color Hotcues</string>
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefdeckdlg.ui]: The spelling ("hotcue" vs. hot cue") is inconsistent and the case is slightly inconsistent with the remaining labels.

Proposal:
Auto hot cue colors   [ ] Assign predefined colors to newly created hot cue points

**Comment by uklotzde** [comment]
Nice addition! Just a minor comment about spelling. A native speaker may double check this.

**Comment by Holzhaus** [comment]
@uklotzde @daschuer Let me know if you want me to change anything else before merge. In particular, should I use a `lowercase_with_underscore` or `PascalCase` name as control object key?

**Comment by uklotzde** [comment]
I think PascalCase is fine since most properties in the [Controls] section already use it.

File: src/preferences/dialog/dlgprefdeck.cpp [hunk 2]
```diff
@@ -166,6 +166,12 @@ DlgPrefDeck::DlgPrefDeck(QWidget * parent, MixxxMainWindow * mixxx,
     connect(checkBoxSeekToCue, SIGNAL(toggled(bool)),
             this, SLOT(slotJumpToCueOnTrackLoadCheckbox(bool)));
 
+    // Automatically assign a color to new hot cues
+    m_bAssignHotcueColors = m_pConfig->getValue(ConfigKey("[Controls]", "AutoHotcueColors"), false);
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefdeck.cpp]: I thought about enabling this feature by default to let users notice. But on the other hand that might confuse and distract existing users. Let's be conservative and let users discover it for themselves.

**Comment by uklotzde** [comment]
LGTM. Thank you very much!

**Comment by Be-ing** [comment]
I don't understand the purpose of this new option. As far as I can tell there is no way for users to specify what hotcue gets assigned to what color unless they do that manually for each hotcue, so what is the benefit of assigning hotcues to random colors with this option?

**Comment by uklotzde** [comment]
The colors help to visually distinguish different hot cues. The assigned color follows the predefined color palette which is reasonable and convenient. If you want to assign different color you have to edit the cue point manually as before.

**Comment by Holzhaus** [comment]
@Be-ing Assigning different colors automatically is what Serato does by default, too. This feature comes in handy when using a hardware controller with colored performance pads, where you don't have a number printed on the pad. IMO it makes it much easier to remember what hot cue is used for what.

Editing a hot cue point to assign a color manually takes quite a bit of time and can be difficult to do with a DJ controller. And sometimes you only need different colors for different hot cues, but do not care which particular color an individual cue point has.

**Pull request merged**